designsafe/static/scripts/geo/controllers/map-sidebar.js

```diff
@@ -0,0 +1,501 @@
+        $timeout( () => {this.map.invalidateSize();}, 10);
+
+        // init an active layer group
+        this.active_layer_group = this.project.layer_groups[0];
+
+        // Auto keep track of current project in the GeoDataService
+        // so that if they switch states they will not lose work...
+        $interval( () => {
+            this.GeoDataService.current_project(this.project);
+        }, 1000);
+
+        this.add_draw_controls(this.active_layer_group.feature_group);
+
+        // This handles making sure that the features that get created with the draw tool
+        // are styled with the default colors etc.
+        this.map.on('draw:created',  (e) => {
+            let object = e.layer;
+            object.options.color = this.settings.default_stroke_color;
+            object.options.fillColor = this.settings.default_fill_color;
+            object.options.fillOpacity = this.settings.default_fill_opacity;
+            this.active_layer_group.feature_group.addLayer(object);
+            if (object instanceof L.Marker) {
+                object.getElement().style.color = this.settings.default_marker_color;
+                object.options.fillColor = this.settings.default_marker_color;
+
+            }
+            this._init_map_layers();
+            this.$scope.$apply();
+        });
+
+        this.map.on('mousemove', (e) => {
+            this.current_mouse_coordinates = e.latlng;
+            this.$scope.$apply();
+        });
+
+        this.$scope.$on('image_popupopen', (e,d) => {
+            // console.log(e, d);
+            this.active_image_marker = d;
+        });
+        this.$scope.$on('image_popupclose', (e,d) => {
+            // console.log(e, d);
+            this.active_image_marker = null;
+        });
+
+    } // end constructor
+
+    fit_map_to_project() {
+        try {
+            this.map.fitBounds(this.project.get_bounds(), { maxZoom: 16 });
+        } catch (e) {
+            console.log('get_bounds fail', e);
+        }
+    }
+
+    add_draw_controls (fg) {
+        let dc = new L.Control.Draw({
+            position: 'topright',
+            draw: {
+
+                marker: {
+                    icon: this.HazmapperDivIcon
+                },
+                circle: false
+            },
+            edit: {
+                featureGroup: fg,
+                remove: true
+            }
+        });
+        this.map.addControl(dc);
+        this.drawControl = dc;
+    }
+
+    feature_click (e) {
+        let layer = e.target;
+        this.current_layer = null;
+        this.project.layer_groups.forEach( (lg)=>{
+            lg.feature_group.getLayers().forEach( (l)=> {
+                if (l == layer) {
+                    l.active = true;
+                } else {
+                    l.active = false;
+                }
+            });
+        });
+        this.$scope.$apply();
+    }
+
+    _init_map_layers() {
```

Comment:
would be cool to actually make these methods "private" like in this example: https://medium.com/front-end-weekly/private-methods-in-es6-and-writing-your-own-db-b2e30866521f

Not really helpful right now but I think this might help us in the long run. Probably worth moving to another ticket.